### PR TITLE
Show number of available skarabs

### PR DESCRIPTION
**Pull request by bxaia**
This PR displays the number of available skarab boards for each CBF proxy. The tooltip displays the number of Available boards out of the total number of boards allocated to that particular CBF proxy.

*Below is the screenshot of the implementation:*
Screenshot 2020-02-04 at 16 17 27

JIRA: [MT-1065](https://skaafrica.atlassian.net/browse/MT-1065)

### CODE REVIEW

File: app/scheduler/subarray-resources/subarray-resources.js [hunk 1]
```diff
@@ -133,6 +133,19 @@
...
         }
 
+        vm.boardsMarkedUp = function (resourceName) {
+            if (!resourceName.startsWith('cbf'))
+                return undefined;
+            return $scope.parent.vm.sensorValues[resourceName + "_boards_marked_up"].value;
+        };
+
+        vm.totalBoards = function (resourceName) {
+            if (resourceName.startsWith('cbf'))
```
Comment (ofaletse) [inline app/scheduler/subarray-resources/subarray-resources.js]: You have block of code not one line
kindly wrap the block of code with braces.

File: app/scheduler/subarray-resources/subarray-resources.js [hunk 2]
```diff
@@ -133,6 +133,19 @@
           return minTimeRemaining;
         }
 
+        vm.boardsMarkedUp = function (resourceName) {
+            if (!resourceName.startsWith('cbf'))
+                return undefined;
+            return $scope.parent.vm.sensorValues[resourceName + "_boards_marked_up"].value;
```
Comment (ofaletse) [inline app/scheduler/subarray-resources/subarray-resources.js]: It appears to me that the sensor `###_boards_marked_up` is not always readily available immediately. you might consider catching that error here.

File: app/scheduler/subarray-resources/subarray-resources.js [hunk 3]
```diff
@@ -133,6 +133,19 @@
...
+        };
+
+        vm.totalBoards = function (resourceName) {
+            if (resourceName.startsWith('cbf'))
+                var boards_marked_standby = $scope.parent.vm.sensorValues[resourceName + "_boards_marked_standby"].value;
+                var boards_marked_up = $scope.parent.vm.sensorValues[resourceName + "_boards_marked_up"].value;
```
Comment (ofaletse) [inline app/scheduler/subarray-resources/subarray-resources.js]: Same comment as above.

File: app/scheduler/subarray-resources/subarray-resources.js (hunk 1 above)
Comment (bxaia) [reply app/scheduler/subarray-resources/subarray-resources.js]: Addressed in the recent commit

File: app/scheduler/subarray-resources/subarray-resources.js (hunk 2 above)
Comment (bxaia) [reply app/scheduler/subarray-resources/subarray-resources.js]: This is also addressed in the latest PR

**Comment by bxaia** [comment]
The screenshot below shows the improvements in the display:

Screenshot 2020-02-05 at 14 52 39

**Comment by bxaia** [comment]
> @bxaia and @bngcebetsha
> Thanks very much for the feature. It looks cool.
> I reviewed your PR and I am requesting some changes before you merge.
> 
> Thanks

addressed in the latest PR

**Comment by bxaia** [comment]
> Nice work. LGTM
> 
> General note: if the layout changes, replace the screen shots in the description with the new one. The description is the first place reviewers will look and it is confusing to have old screenshots.

Thanks I've changed the screenshot on the description.

**Pull request merged**